sourcemash/categorize.py

```diff
@@ -1,23 +1,99 @@
+    "hers", "herself", "him", "himself", "his", "how", "however", "hundred", "ie", \
+    "if", "in", "inc", "indeed", "interest", "into", "is", "it", "its", "itself", \
+    "keep", "last", "latter", "latterly", "least", "less", "ltd", "made", "many", \
+    "may", "me", "meanwhile", "might", "mill", "mine", "more", "moreover", "most", \
+    "mostly", "move", "much", "must", "my", "myself", "name", "namely", "neither", \
+    "never", "nevertheless", "next", "nine", "no", "nobody", "none", "noone", "nor", \
+    "not", "nothing", "now", "nowhere", "of", "off", "often", "on", "once", "one", \
+    "only", "onto", "or", "other", "others", "otherwise", "our", "ours", "ourselves", \
+    "out", "over", "own","part", "per", "perhaps", "please", "put", "rather", "re", \
+    "same", "see", "seem", "seemed", "seeming", "seems", "serious", "several", "she", \
+    "should", "show", "side", "since", "sincere", "six", "sixty", "so", "some", \
+    "somehow", "someone", "something", "sometime", "sometimes", "somewhere", "still", \
+    "such", "system", "take", "ten", "than", "that", "the", "their", "them", \
+    "themselves", "then", "thence", "there", "thereafter", "thereby", "therefore", \
+    "therein", "thereupon", "these", "they", "thickv", "thin", "third", "this", \
+    "those", "though", "three", "through", "throughout", "thru", "thus", "to", \
+    "together", "too", "top", "toward", "towards", "twelve", "twenty", "two", \
+    "un", "under", "until", "up", "upon", "us", "very", "via", "was", "we", "well", \
+    "were", "what", "whatever", "when", "whence", "whenever", "where", "whereafter", \
+    "whereas", "whereby", "wherein", "whereupon", "wherever", "whether", "which", \
+    "while", "whither", "who", "whoever", "whole", "whom", "whose", "why", "will", \
+    "with", "within", "without", "would", "yet", "you", "your", "yours", "yourself", \
+    "yourselves", "the" ] # Source: http://xpo6.com/list-of-english-stop-words/
 
-	# Get word counts from title, text of article
-	categories.update(title + text)
+class Categorizer:
 
-	# Give weights to categories found in title
-	for word in title:
-		categories[word] *= 2
+    def __init__(self):
+        self.title_categories = Counter()
 
-	# Return top 2 words in categories
-	most_frequent = categories.most_common(2)
+    def categorize_item(self, title, text):
+        cat1 = ""
+        cat2 = ""
+        categories = Counter()
+
+        # Only consider words present in category dictonary
+        title = filter(lambda word: word in self.title_categories, title.split())
+        text = filter(lambda word: word in self.title_categories, text.split())
+        title = [word.strip(punctuation) for word in title]
+        text = [word.strip(punctuation) for word in text]
+
+        # Get word counts from title, text of article
+        categories.update(title + text)
+
+        # Give weights to categories found in title
+        for word in title:
+            categories[word] *= 2
```

Comment:
Extract '2' out to CONSTANT, like "TITLE_WORD_WEIGHT," or something of that nature.